eox_hooks/tasks.py

```diff
@@ -0,0 +1,72 @@
+"""
+Task file for eox-hooks plugin.
+"""
+import logging
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.sites.models import Site
+
+try:
+    from eox_core.edxapp_wrapper.users import get_course_enrollment, get_edxapp_user
+    from eox_core.edxapp_wrapper.coursekey import get_valid_course_key
+    from course_modes.models import CourseMode
+except ImportError:
+    get_course_enrollment = lambda: object
+    get_edxapp_user = lambda: object
+    CourseMode = object
+
+ENROLLMENT_MAX_RETRIES = 3
+CourseEnrollment = get_course_enrollment()
+log = logging.getLogger(__name__)
+
+
+@shared_task(bind=True, max_retries=ENROLLMENT_MAX_RETRIES)
```

Comment:
you could use `autoretry_for`  instead of caching the exception

Reply:
autoretry is available in celery 4.0

Reply:
how sad